### DISCUSSION TITLE
Focus in FieldArray is not working

### DISCUSSION THREAD

**Discussion by radokristof** [react-hook-form/react-hook-form#7873] [Q&A] [answered]
I'm trying to use focus when appending an element to a `FieldArray` but it does not work as it should.

I try adding an element like this:

```
append({ firstName: "appendBill", lastName: "appendLuo" }, {focusName: `test[${fields.length - 1}].lastName`});
```

Based on this codesandbox:

https://codesandbox.io/s/vy8fv?file=/src/index.js

(I'm using the latest v7 version, but I did not find a good working example in codesandbox where I could show it, but anyways, it doesn't work in either version).

In this discussion it was proposed to be added in the v6 version:
https://github.com/react-hook-form/react-hook-form/discussions/1686

So I assume it should work in both version (also these functions are not really well documented on the page).

Thanks for your help!

**Comment by Moshyfawn** [comment, marked as answer]
I believe, your issue is you're not using the dot notation to set the `focusName`. 
```js
append(
  { firstName: "John", lastName: "Doe" },
  {
    focusName: "fields.0.lastName"
  }
)
```
See [this working CSB example](https://codesandbox.io/s/gifted-forest-1ibdt1?file=/src/App.tsx:1384-1550)

**Reply by Moshyfawn** [reply]
But you're right, there's not much in docs about the array focus options.
@bluebill1049 I can work on adding some info on these

**Reply by bluebill1049** [reply]
lol, I am just starting to open a codesandbox and you already show a working example 👍 thank you @Moshyfawn. yes plz.

**Reply by radokristof** [reply]
Thanks for both of you! This was clearly an issue, that I was using an array notation instead of the dot notation (didn't knew where I was looking, this was clearly in the examples).

~~However it is still not working for me (I'm using MUI with Controller), but I will try to debug this or show a working example (where it is actually not working with MUI).~~

Well, the problem was that I was not using dot notation for registering the inputs as well. I was using an an array notation.
I wonder how this worked at all then...
